### PR TITLE
picgo: Add autoupdate

### PR DESCRIPTION
**Pull request by Gladtbam**

### CODE REVIEW

**Comment by rashil2000** [comment]
Add autoupdate too. See https://github.com/ScoopInstaller/Extras/commit/9d2f42d1679a8a61d6b827d3e78ef553016e923c#diff-5d13722ca8a08c50336c43f12427760d5bedc4599358b8e0c92cd3594062ddfd for reference.

File: bucket/picgo.json [hunk 1]
```diff
@@ -23,5 +23,16 @@
     "checkver": {
         "url": "https://github.com/Molunerfinn/PicGo/releases",
         "regex": "/PicGo-Setup-([\\d.]+)-x64\\.exe"
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://github.com/Molunerfinn/PicGo/releases/download/v$version/PicGo-Setup-$version.exe#/dl.7z",
```
Comment (rashil2000) [inline bucket/picgo.json:30]:
```suggestion
                "url": "https://github.com/Molunerfinn/PicGo/releases/download/v$version/PicGo-Setup-$version-x64.exe#/dl.7z",
```

**Review by rashil2000 — CHANGES_REQUESTED** [review]
They now provide 32bit release. Add that too. https://github.com/Molunerfinn/PicGo/releases/download/v2.3.0/PicGo-Setup-2.3.0-ia32.exe

**Comment by Gladtbam** [comment]
Their latest.yml file does not provide a hash of x32bit.

**Comment by rashil2000** [comment]
> Their latest.yml file does not provide a hash of x32bit.

The latest.yml file for their next beta release `2.3.1-beta.2` contains hash for 32bit. Looks like they forgot to do that for `2.3.0`. Can you open an issue there to remind them to include it?

File: bucket/picgo.json [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
             "pre_install": [
                 "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\app-64.7z\" \"$dir\"",
                 "Remove-Item \"$dir\\`$*\", \"$dir\\Uninstall*\" -Recurse"
             ]
+        },
+        "32bit": {
+            "url": "https://github.com/Molunerfinn/PicGo/releases/download/v2.3.0/PicGo-Setup-2.3.0-ia32.exe#/dl.7z",
+            "hash": "sha512:3fe915b42be565a20005cb94361dca0d9525f38b18221399da51ab79acbb91d7627bd0d988360fea16f592b8db0cd668e3626562224c91276626c87e135d7625",
+            "pre_install": [
+                "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\app-ia32.7z\" \"$dir\"",
```
Comment (rashil2000) [inline bucket/picgo.json:19]:
```suggestion
                "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\app-32.7z\" \"$dir\"",
```

You should always confirm/check installation for 32bit by running `scoop install <filename>.json -a 32bit`

**Review by rashil2000 — CHANGES_REQUESTED** [review]

**Comment by Gladtbam** [comment]
Now, they added hash for 32bit. But I  do not understand how to write `regex` for 32bit.
As you can see, I just copy-pasted for now

**Comment by rashil2000** [comment]
Check if this works `(?sm)ia32\\.exe?.*$base64`

**Comment by Gladtbam** [comment]
There is one thing I don't quite understand. Why the obtained hash is `hash 256` instead of `hash 512`
image

**Comment by rashil2000** [comment]
Because they provide sha512 in the yml file. If it was sha256, we'd get hash 256

**Comment by Gladtbam** [comment]
> Because they provide sha512 in the yml file. If it was sha256, we'd get hash 256
> 
> Also, you can comment `/verify` to see if your changes pass 

OK, I may have misinterpreted what I said.

They provided hash 512, but got hash 256 by checkver.ps1 instead. Why?

**Comment by Gladtbam** [comment]
OK! I deleted `"url": "$baseurl/latest.yml"`. Now it works fine.

**Comment by niheaven** [comment]
You now has wrong au.hash...

**Comment by niheaven** [comment]
```json
"hash": {
    "url": "$baseurl/latest.yml",
    "regex": "(?s)$basename.*?$sha512"
}
```

This should work, put it outsides au arch, it's noarch hash extraction script.

**Comment by Gladtbam** [comment]
> You now has wrong au.hash...

No. In fact, it gets the correct `hash 256`.

实际上得到的hash 256是正确的，它通过将文件下载下来得到的

**Comment by Gladtbam** [comment]
> ```json
> "hash": {
>     "url": "$baseurl/latest.yml",
>     "regex": "(?s)$basename.*?$sha512"
> }
> ```
> 
> This should work, put it outsides au arch, it's noarch hash extraction script.

use it, PowerShell reply `Could not find hash in *.../latest.yml`
And reply `Downloading ...exe to compute hashes`

用这个scoop报在latest.yml文件中找不到hash，然后会下载文件计算hash 256

**Comment by issaclin32** [comment]
The checksum is encoded in Base64 (`sha512: yB7AmmEt6Qw5gtxQjtgWIHAsV3oyiPoVrfcd6gre5/428w+S7UoaWmTVyXH6xo9gO9HNTlUcOt90I2DE2ha8Dg==`)
It will work if we use **$base64**.

```
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://github.com/Molunerfinn/PicGo/releases/download/v$version/PicGo-Setup-$version-x64.exe#/dl.7z"
            },
            "32bit": {
                "url": "https://github.com/Molunerfinn/PicGo/releases/download/v$version/PicGo-Setup-$version-ia32.exe#/dl.7z"
            }
        },
        "hash": {
            "url": "$baseurl/latest.yml",
            "regex": "(?s)$basename.*?$base64"
        }
    }
```

**Comment by niheaven** [comment]
Yes, it should be `$base64`. I typed the code by phone and made a mistake.

**Comment by Gladtbam** [comment]
> The checksum is encoded in Base64 (`sha512: yB7AmmEt6Qw5gtxQjtgWIHAsV3oyiPoVrfcd6gre5/428w+S7UoaWmTVyXH6xo9gO9HNTlUcOt90I2DE2ha8Dg==`) It will work if we use **$base64**.
> 
> ```
>     "autoupdate": {
>         "architecture": {
>             "64bit": {
>                 "url": "https://github.com/Molunerfinn/PicGo/releases/download/v$version/PicGo-Setup-$version-x64.exe#/dl.7z"
>             },
>             "32bit": {
>                 "url": "https://github.com/Molunerfinn/PicGo/releases/download/v$version/PicGo-Setup-$version-ia32.exe#/dl.7z"
>             }
>         },
>         "hash": {
>             "url": "$baseurl/latest.yml",
>             "regex": "(?s)$basename.*?$base64"
>         }
>     }
> ```

Yes, this's OK!

**Comment by niheaven** [comment]
You should use one au.hash, just likes @issaclin32 comment.

**Comment by Gladtbam** [comment]
I might know why I get it wrong when I use `(?s)$basename.*?$base64`
Thanks

**Comment by niheaven** [comment]
You put `(?s)` in wrong place, right? 😁

**Review by niheaven — APPROVED** [review]

**Comment by Gladtbam** [comment]
You may not believe it, it parses the correct md5😭

**Pull request merged**